tests/hil/tests/ota/test.c

```diff
@@ -160,14 +169,118 @@ static void test_block_ops(void)
         }
         else
         {
-            GLTH_LOGI(TAG, "Received block %d", block_index);
+            GLTH_LOGI(TAG, "Received block %zu", block_index);
             GLTH_LOGI(TAG, "is_last: %d", is_last);
         }
 
         ++block_index;
     }
 }
 
+enum golioth_status write_artifact_block_cb(const struct golioth_ota_component *component,
+                                            uint32_t block_idx,
+                                            uint8_t *block_buffer,
+                                            size_t block_buffer_len,
+                                            bool is_last,
+                                            size_t negotiated_block_size,
+                                            void *arg)
+{
+    if (!arg)
+    {
+        GLTH_LOGE(TAG, "arg is NULL but should be a resume_ctx");
+        return GOLIOTH_ERR_INVALID_FORMAT;
+    }
+
+    struct resume_ctx *ctx = (struct resume_ctx *) arg;
+
+    if (block_idx == 0)
+    {
+        /* This was already done with the create() function
+         * but let's call it again to test the api
+         */
+        golioth_sys_sha256_init(ctx->sha);
```

Comment:
Per my comment on the earlier PR, I don't think it makes sense to have both `create()` and `init()` for the purposes of our port layer.